go-controller/pkg/cni/helper_linux.go

```diff
@@ -252,60 +252,76 @@ func setupInterface(netns ns.NetNS, containerID, ifName string, ifInfo *PodInter
 	return hostIface, contIface, nil
 }
 
+// generate a unique interface name for the temporary netdev that will be moved to pod namespace
+// the generated ifname suffix will be "_<ifName>"
+func generateIfName(containerID string, ifName string) string {
+	ifnameSuffix := "_" + ifName
+	// ifname max length is 15
+	return containerID[:(15-len(ifnameSuffix))] + ifnameSuffix
+}
+
+// generate a unique interface name for the port representor
+// the generated ifname suffix will be "_r_<ifName>"
+func generateRepresentorIfName(containerID string, ifName string) string {
+	ifnameSuffix := "_r_" + ifName
+	// ifname max length is 15
+	return containerID[:(15-len(ifnameSuffix))] + ifnameSuffix
+}
+
 // Setup sriov interface in the pod
 func setupSriovInterface(netns ns.NetNS, containerID, ifName string, ifInfo *PodInterfaceInfo, deviceID string) (*current.Interface, *current.Interface, error) {
 	hostIface := &current.Interface{}
 	contIface := &current.Interface{}
-	ifnameSuffix := ""
-
 	netdevice := ifInfo.NetdevName
 
 	// 1. Move netdevice to Container namespace
-	err := moveIfToNetns(netdevice, netns)
-	if err != nil {
-		return nil, nil, err
-	}
-
-	err = netns.Do(func(hostNS ns.NetNS) error {
-		contIface.Name = ifName
-		err = renameLink(netdevice, contIface.Name)
-		if err != nil {
-			return err
-		}
-		link, err := util.GetNetLinkOps().LinkByName(contIface.Name)
+	if len(netdevice) != 0 {
```

Comment:
may I ask when netdevice can be empty string?

Reply:
I've done this because of the next PR in mind. In case of vhost-vdpa devices, there is no netdev to be moved to the pod namespace, a character device is mounted in the pod instead (e.g., /dev/vhost-vdpa-0). So in this case the function `util.GetNetdevNameFromDeviceId` will return an empty netdev (https://github.com/ovn-org/ovn-kubernetes/blob/e0f97cf6b6a4cd2e06d0a19184ae92674a1ef845/go-controller/pkg/cni/cni.go#L115). I don't think this change can cause any harm, but happy to remove it if you want.

Reply:
It's unclear to me what will happen when running on a DPU cluster. I'm wondering if it makes more sense to return an error that is signifying that no netdev needs to be moved. I'm wondering if returning an empty string here is more a hack to signify that the user should not have called this function in the first place (when using vpda).

Reply:
The DPU case is unclear to me as well, maybe someone else can jump in on it? Anyway this use case is not supported for vdpa yet, maybe in the future. The mentioned function returns a netdev from the PCI address and unfortunately we don't know upfront if this is the vhost-vdpa case or not. In the case of virtio-vdpa, a valid netdev is returned instead. Also, not sure it would be an error condition because the normal flow of the program is kept unaltered.